tests/recovery/recovery_test/recovery_test.go

```diff
@@ -168,6 +172,172 @@ func TestSnapshotFixInvalidFiles(t *testing.T) {
+
+	// add files
+	fileSize := 1 * 1024 * 1024
+	numFiles := 200
+
+	err = bm.GenerateRandomFiles(fileSize, numFiles)
+	require.NoError(t, err)
+
+	err = bm.FileHandler.CopyAllFiles(bm.PathToTakeSnapshot, copyDir)
+	require.NoError(t, err)
+
+	// modify original files
+	content := "\nthis is a test for TestConsistencyWhenKill9AfterModify\n"
+	err = bm.FileHandler.ModifyDataSetWithContent(copyDir, content)
+	require.NoError(t, err)
+	log.Println("Copy content and modify the files.")
+
+	// kill the kopia command before it exits
+	kopiaExe := os.Getenv("KOPIA_EXE")
+
+	cmd := exec.Command(kopiaExe, "repo", "connect", "filesystem", "--path="+dataRepoPath, "--content-cache-size-mb", "500", "--metadata-cache-size-mb", "500", "--no-check-for-updates")
+	env := []string{"KOPIA_PASSWORD=" + testenv.TestRepoPassword}
+	cmd.Env = append(os.Environ(), env...)
+
+	o, err := cmd.CombinedOutput()
+	require.NoError(t, err)
+	log.Println(string(o))
+
+	cmd = exec.Command(kopiaExe, "snap", "create", copyDir, "--json", "--parallel=1")
+
+	log.Println("Kill the kopia command before it exits:")
+	killOnCondition(t, cmd)
+
+	log.Println("Verify snapshot corruption:")
+	// verify snapshot corruption
+	err = bm.VerifySnapshot()
+	require.NoError(t, err)
+
+	// Create a temporary dir to restore a snapshot
+	restoreDir := t.TempDir()
+	if restoreDir == "" {
+		t.FailNow()
+	}
+
+	// try to restore a snapshot without any error messages.
+	stdout, err := bm.RestoreGivenOrRandomSnapshot(snapID, restoreDir)
+	require.NoError(t, err)
+
+	log.Println(stdout)
+
+	log.Println("Compare restored data and original data:")
+	require.NoError(t, bm.FileHandler.CompareDirs(restoreDir, cmpDir))
+}
+
+func killOnCondition(t *testing.T, cmd *exec.Cmd) {
+	t.Helper()
+
+	stderrPipe, err := cmd.StderrPipe()
+	require.NoError(t, err)
+
+	// execute kill -9 while receive ` | 1 hashing, 0 hashed (65.5 KB), 0 cached (0 B), uploaded 0 B, estimating...` message
+	var wg sync.WaitGroup
+
+	var mu sync.Mutex
+
+	// Add a WaitGroup counter for the first goroutine
+	wg.Add(1)
+
+	errOut := bytes.Buffer{}
+
+	go func() {
+		mu.Lock()
+		defer mu.Unlock()
+		defer wg.Done()
+
+		// Create a scanner to read from stderrPipe
+		scanner := bufio.NewScanner(stderrPipe)
+		for scanner.Scan() {
+			output := scanner.Text()
+			log.Println(output)
+			errOut.Write(scanner.Bytes())
+			errOut.WriteByte('\n')
+
+			log.Println(output)
+			// Check if the output contains the "hashing" etc.
+			if strings.Contains(output, "hashing") && strings.Contains(output, "hashed") && strings.Contains(output, "uploaded") || strings.Contains(output, "Snapshotting") {
+				log.Println("Detaching and terminating target process")
+				cmd.Process.Kill()
+
+				break
+			}
+		}
+	}()
+
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		panic(err)
+	}
+
+	wg.Add(1)
+
+	o := bytes.Buffer{}
+
+	go func() {
+		defer wg.Done()
+
+		// Create a scanner to read from stdoutPipe
+		scanner := bufio.NewScanner(stdoutPipe)
```

Comment:
Do you really need to check BOTH stdout and stderr pipes ? I think you should be able to drop one of them.

Reply:
I believe it's still necessary to have the stdoutpipe because there's a possibility that the snapshot will be created successfully. Although it's unlikely for the stdoutpipe to be executed in such cases, there are scenarios where the test environment is not cleaned properly in pipeline and the previous dataset is used for running the test. In such cases, it's important to know whether the snapshot was created successfully, even though the occurrence of this situation is unlikely.

Reply:
It should not matter whether if snapshot was created successfully. If Kopia prints to stderr then it will always print that line to stderr even if it succeeded.

Reply:
I decide to delete stdoutPipe because the log about uploading only print at stderrPipe.